commit 8f8a088c2127c729638da8f2d33860e346c01eda
Author: Martin Willi <martin@strongswan.org>
Date:   Wed Nov 25 00:29:53 2009 +0000

    xfrm: Use the user specified truncation length in ESP and AH
    
    Instead of using the hardcoded truncation for authentication
    algorithms, use the truncation length specified on xfrm_state.
    
    Signed-off-by: Martin Willi <martin@strongswan.org>
    Signed-off-by: David S. Miller <davem@davemloft.net>

diff --git a/net/ipv4/esp4.c b/net/ipv4/esp4.c
index 12f7287e902d..1948895beb6d 100644
--- a/net/ipv4/esp4.c
+++ b/net/ipv4/esp4.c
@@ -530,7 +530,7 @@ static int esp_init_authenc(struct xfrm_state *x)
 		}
 
 		err = crypto_aead_setauthsize(
-			aead, aalg_desc->uinfo.auth.icv_truncbits / 8);
+			aead, x->aalg->alg_trunc_len / 8);
 		if (err)
 			goto free_key;
 	}